Index: org/columba/core/i18n/global/global.properties
===================================================================
RCS file: /cvsroot/columba/columba/org/columba/core/i18n/global/global.properties,v
retrieving revision 1.1
diff -u -3 -p -c -w -r1.1 global.properties
*** org/columba/core/i18n/global/global.properties	3 Sep 2002 10:45:33 -0000	1.1
--- org/columba/core/i18n/global/global.properties	17 Sep 2002 15:47:42 -0000
*************** menu_edit_selectall = Select All
*** 63,70 ****
  menu_edit_selectall_mnemonic=a
  menu_edit_replace = Replace ...

!
!
!
!
!
--- 63,68 ----
  menu_edit_selectall_mnemonic=a
  menu_edit_replace = Replace ...

! # 09/16/02 ALP
! # Added as part of external editor support
! menu_edit_extern_edit = Use External Editor
Index: org/columba/modules/mail/gui/composer/ComposerMenu.java
===================================================================
RCS file: /cvsroot/columba/columba/org/columba/modules/mail/gui/composer/ComposerMenu.java,v
retrieving revision 1.7
diff -u -3 -p -c -w -r1.7 ComposerMenu.java
*** org/columba/modules/mail/gui/composer/ComposerMenu.java	3 Sep 2002 10:45:32 -0000	1.7
--- org/columba/modules/mail/gui/composer/ComposerMenu.java	17 Sep 2002 15:47:42 -0000
*************** public class ComposerMenu extends JMenuB
*** 155,160 ****
--- 155,182 ----

  		editMenu.addSeparator();

+
+         // 09/16/02 ALP
+         // Added as part of external editor support
+ 		menuItem =
+ 			new JMenuItem(
+                 MailResourceLoader.getString("menu","composer","menu_edit_extern_edit"));
+
+ // 		menuItem.setMnemonic(KeyEvent.VK_G);
+ // 		menuItem.setAccelerator(
+ // 			KeyStroke.getKeyStroke(KeyEvent.VK_G, ActionEvent.CTRL_MASK));
+ 		menuItem.setEnabled(true);
+ 		menuItem.setActionCommand("EXTERNEDIT");
+ // 		menuItem.addActionListener(
+ // 			MainInterface.mainFrame.getActionListener() );
+ 		menuItem.addActionListener(composerInterface.composerActionListener.externEditAction);
+
+ 		editMenu.add(menuItem);
+
+ 		editMenu.addSeparator();
+         // 09/16/02 ALP
+         // End addition
+
  		menuItem =
  			new JMenuItem(
  				MailResourceLoader.getString("menu","mainframe","menu_preferences_general")); //$NON-NLS-1$ //$NON-NLS-2$
Index: org/columba/modules/mail/gui/composer/action/ComposerActionListener.java
===================================================================
RCS file: /cvsroot/columba/columba/org/columba/modules/mail/gui/composer/action/ComposerActionListener.java,v
retrieving revision 1.22
diff -u -3 -p -c -w -r1.22 ComposerActionListener.java
*** org/columba/modules/mail/gui/composer/action/ComposerActionListener.java	3 Sep 2002 10:45:33 -0000	1.22
--- org/columba/modules/mail/gui/composer/action/ComposerActionListener.java	17 Sep 2002 15:47:42 -0000
*************** import org.columba.modules.mail.gui.comp
*** 33,38 ****
--- 33,43 ----
  import org.columba.modules.mail.util.MailResourceLoader;
  //import sun.security.krb5.internal.crypto.e;

+
+ // 09/16/02 ALP
+ // Added as part of external editor support
+ import org.columba.modules.mail.gui.composer.util.ExternalEditor;
+
  public class ComposerActionListener implements ActionListener {
  	private ComposerInterface composerInterface;

*************** public class ComposerActionListener impl
*** 57,62 ****
--- 62,70 ----
  	public BasicAction selectAllAction;
  	public BasicAction signAction;
  	public BasicAction encryptAction;
+     // 09/16/02 ALP
+     // Added as part of external editor support
+ 	public BasicAction externEditAction;

  	public ComposerActionListener(ComposerInterface iface) {
  		composerInterface = iface;
*************** public class ComposerActionListener impl
*** 357,362 ****
--- 365,384 ----
  		encryptAction.addActionListener(this);
  		encryptAction.setEnabled(true);

+         // 09/16/02 ALP
+         // Added as part of external editor support
+ 		externEditAction =
+ 			new BasicAction(
+ 				MailResourceLoader.getString("menu","composer", "menu_edit_extern_edit"),
+ 				MailResourceLoader.getString("menu","composer", "menu_edit_extern_edit"),
+ 				MailResourceLoader.getString("menu","composer", "menu_edit_extern_edit"),
+ 				"EXTERNEDIT",
+ 				ImageLoader.getSmallImageIcon("extern_edit_small.png"),
+ 				null,
+ 				MailResourceLoader.getMnemonic("menu","composer", "menu_edit_extern_edit"),
+ 				null);
+ 		externEditAction.addActionListener(this);
+ 		externEditAction.setEnabled(true);
  	}

  	protected boolean checkInformation() {
*************** public class ComposerActionListener impl
*** 577,581 ****
--- 599,613 ----

  		}

+         // 09/16/02 ALP
+         // Added as part of external editor support
+         if (command.equals("EXTERNEDIT")) {
+           System.out.println( "********************************************" );
+           System.out.println( "*** External Editor support is pre-alpha ***" );
+           System.out.println( "*** USE AT YOUR OWN RISK                 ***" );
+           System.out.println( "********************************************" );
+           ExternalEditor Ed = new ExternalEditor();
+           Ed.startExternalEditor(composerInterface.editorController.getView());
+         }
  	}
  }
Index: org/columba/modules/mail/gui/mimetype/ColumbaViewer.java
===================================================================
RCS file: /cvsroot/columba/columba/org/columba/modules/mail/gui/mimetype/ColumbaViewer.java,v
retrieving revision 1.3
diff -u -3 -p -c -w -r1.3 ColumbaViewer.java
*** org/columba/modules/mail/gui/mimetype/ColumbaViewer.java	6 Jun 2002 09:19:10 -0000	1.3
--- org/columba/modules/mail/gui/mimetype/ColumbaViewer.java	17 Sep 2002 15:47:42 -0000
*************** public class ColumbaViewer extends Defau
*** 21,27 ****
  {


!     public void openWith( MimeHeader header, File tempFile )
      {

          boolean save = false;
--- 21,29 ----
  {


!     Process Child;
!
!     public Process openWith( MimeHeader header, File tempFile )
      {

          boolean save = false;
*************** public class ColumbaViewer extends Defau
*** 41,56 ****
  	//System.out.println("cmd:"+cmd);

          try {
!             Runtime.getRuntime().exec( cmd );
          }
          catch( IOException e ) {
              JOptionPane.showMessageDialog( MainInterface.mainFrame,
                                             "Error: "+e.getMessage() );
                //System.out.println( e );
          }
      }

!     public void open( MimeHeader header, File tempFile )
      {
          String viewer = MimeRouter.getViewer( header );

--- 43,59 ----
  	//System.out.println("cmd:"+cmd);

          try {
!           Child = Runtime.getRuntime().exec( cmd );
          }
          catch( IOException e ) {
              JOptionPane.showMessageDialog( MainInterface.mainFrame,
                                             "Error: "+e.getMessage() );
                //System.out.println( e );
          }
+         return Child;
      }

!     public Process open( MimeHeader header, File tempFile )
      {
          String viewer = MimeRouter.getViewer( header );

*************** public class ColumbaViewer extends Defau
*** 70,86 ****


          try {
!             Runtime.getRuntime().exec( viewer+" "+tempFile.toString() );
          }
          catch( IOException e ) {
              JOptionPane.showMessageDialog( MainInterface.mainFrame,
                                             "Error: "+e.getMessage() );
                //System.out.println( e );
          }
      }


!     public void openUrl( String url )
      {
          boolean save = false;
          String viewer;
--- 73,90 ----


          try {
!           Child = Runtime.getRuntime().exec( viewer+" "+tempFile.toString() );
          }
          catch( IOException e ) {
              JOptionPane.showMessageDialog( MainInterface.mainFrame,
                                             "Error: "+e.getMessage() );
                //System.out.println( e );
          }
+         return Child;
      }


!     public Process openUrl( String url )
      {
          boolean save = false;
          String viewer;
*************** public class ColumbaViewer extends Defau
*** 103,109 ****

          try
          {
!             Runtime.getRuntime().exec( viewer+" "+url );
          }
          catch( Exception ex )
          {
--- 107,113 ----

          try
          {
!           Child = Runtime.getRuntime().exec( viewer+" "+url );
          }
          catch( Exception ex )
          {
*************** public class ColumbaViewer extends Defau
*** 111,119 ****
              JOptionPane.showMessageDialog( MainInterface.mainFrame,
                                                     "Error: "+ex.getMessage() );
          }
      }

!     public void openWithUrl( String url )
      {
          boolean save = false;
          String viewer;
--- 115,124 ----
              JOptionPane.showMessageDialog( MainInterface.mainFrame,
                                                     "Error: "+ex.getMessage() );
          }
+         return Child;
      }

!     public Process openWithUrl( String url )
      {
          boolean save = false;
          String viewer;
*************** public class ColumbaViewer extends Defau
*** 132,138 ****

          try
          {
!             Runtime.getRuntime().exec( viewer+" "+url );
          }
          catch( Exception ex )
          {
--- 137,143 ----

          try
          {
!           Child = Runtime.getRuntime().exec( viewer+" "+url );
          }
          catch( Exception ex )
          {
*************** public class ColumbaViewer extends Defau
*** 140,147 ****
              JOptionPane.showMessageDialog( MainInterface.mainFrame,
                                                     "Error: "+ex.getMessage() );
          }
      }

!
!
  }
--- 145,154 ----
              JOptionPane.showMessageDialog( MainInterface.mainFrame,
                                                     "Error: "+ex.getMessage() );
          }
+         return Child;
      }

!   public Process getChild(){
!     return Child;
!   }
  }
Index: org/columba/modules/mail/gui/mimetype/DefaultViewer.java
===================================================================
RCS file: /cvsroot/columba/columba/org/columba/modules/mail/gui/mimetype/DefaultViewer.java,v
retrieving revision 1.3
diff -u -3 -p -c -w -r1.3 DefaultViewer.java
*** org/columba/modules/mail/gui/mimetype/DefaultViewer.java	6 Jun 2002 09:19:10 -0000	1.3
--- org/columba/modules/mail/gui/mimetype/DefaultViewer.java	17 Sep 2002 15:47:42 -0000
*************** import java.io.*;
*** 17,28 ****

  public abstract class DefaultViewer
  {

!     public abstract void openWith( MimeHeader header, File tempFile );

!     public abstract void open( MimeHeader header, File tempFile );

!     public abstract void openUrl( String url );


  }
\ No newline at end of file
--- 17,32 ----

  public abstract class DefaultViewer
  {
+     Process Child;

!     public abstract Process openWith( MimeHeader header, File tempFile );

!     public abstract Process open( MimeHeader header, File tempFile );

!     public abstract Process openUrl( String url );

+   public Process getChild(){
+     return Child;
+   }

  }
Index: org/columba/modules/mail/gui/mimetype/MimeTypeViewer.java
===================================================================
RCS file: /cvsroot/columba/columba/org/columba/modules/mail/gui/mimetype/MimeTypeViewer.java,v
retrieving revision 1.3
diff -u -3 -p -c -w -r1.3 MimeTypeViewer.java
*** org/columba/modules/mail/gui/mimetype/MimeTypeViewer.java	6 Jun 2002 09:19:10 -0000	1.3
--- org/columba/modules/mail/gui/mimetype/MimeTypeViewer.java	17 Sep 2002 15:47:42 -0000
*************** import java.io.*;
*** 20,25 ****
--- 20,26 ----
  public class MimeTypeViewer
  {
      private String osName;
+   Process Child;

      public MimeTypeViewer()
      {
*************** public class MimeTypeViewer
*** 39,45 ****
          return false;
      }

!     public void openWith( MimeHeader header, File tempFile )
      {
          if ( isWindowsPlatform() == true )
          {
--- 40,46 ----
          return false;
      }

!     public Process openWith( MimeHeader header, File tempFile )
      {
          if ( isWindowsPlatform() == true )
          {
*************** public class MimeTypeViewer
*** 49,59 ****
          else
          {
              ColumbaViewer viewer = new ColumbaViewer();
!             viewer.openWith( header, tempFile );
          }
      }

!     public void open( MimeHeader header, File tempFile )
      {
          if ( isWindowsPlatform() == true )
          {
--- 50,61 ----
          else
          {
              ColumbaViewer viewer = new ColumbaViewer();
!             Child = viewer.openWith( header, tempFile );
          }
+         return Child;
      }

!     public Process open( MimeHeader header, File tempFile )
      {
          if ( isWindowsPlatform() == true )
          {
*************** public class MimeTypeViewer
*** 63,74 ****
          else
          {
              ColumbaViewer viewer = new ColumbaViewer();
!             viewer.open( header, tempFile );
          }
      }


!     public void openUrl( String url )
      {
          if ( isWindowsPlatform() == true )
          {
--- 65,77 ----
          else
          {
              ColumbaViewer viewer = new ColumbaViewer();
!             Child = viewer.open( header, tempFile );
          }
+         return Child;
      }


!     public Process openUrl( String url )
      {
          if ( isWindowsPlatform() == true )
          {
*************** public class MimeTypeViewer
*** 78,88 ****
          else
          {
              ColumbaViewer viewer = new ColumbaViewer();
!             viewer.openUrl( url );
          }
      }

!     public void openWithUrl( String url )
      {
           if ( isWindowsPlatform() == true )
          {
--- 81,92 ----
          else
          {
              ColumbaViewer viewer = new ColumbaViewer();
!             Child = viewer.openUrl( url );
          }
+         return Child;
      }

!     public Process openWithUrl( String url )
      {
           if ( isWindowsPlatform() == true )
          {
*************** public class MimeTypeViewer
*** 92,100 ****
          else
          {
              ColumbaViewer viewer = new ColumbaViewer();
!             viewer.openWithUrl( url );
          }
!
      }


--- 96,104 ----
          else
          {
              ColumbaViewer viewer = new ColumbaViewer();
!             Child = viewer.openWithUrl( url );
          }
!         return Child;
      }


*************** public class MimeTypeViewer
*** 129,135 ****
          }

      }
-
-
-
  }
--- 133,136 ----
Index: org/columba/modules/mail/gui/mimetype/WindowsViewer.java
===================================================================
RCS file: /cvsroot/columba/columba/org/columba/modules/mail/gui/mimetype/WindowsViewer.java,v
retrieving revision 1.4
diff -u -3 -p -c -w -r1.4 WindowsViewer.java
*** org/columba/modules/mail/gui/mimetype/WindowsViewer.java	6 Jun 2002 09:19:10 -0000	1.4
--- org/columba/modules/mail/gui/mimetype/WindowsViewer.java	17 Sep 2002 15:47:42 -0000
*************** import java.io.*;
*** 18,34 ****
  public class WindowsViewer extends DefaultViewer
  {

!     public void openWith( MimeHeader header, File tempFile )
      {
          openDocument( tempFile.getPath() );
      }

!     public void open( MimeHeader header, File tempFile )
      {
          openDocument( tempFile.getPath() );
      }

!     public void openUrl( String url )
      {
          String osName = System.getProperty("os.name" );

--- 18,36 ----
  public class WindowsViewer extends DefaultViewer
  {

!     public Process openWith( MimeHeader header, File tempFile )
      {
          openDocument( tempFile.getPath() );
+         return Child;
      }

!     public Process open( MimeHeader header, File tempFile )
      {
          openDocument( tempFile.getPath() );
+         return Child;
      }

!     public Process openUrl( String url )
      {
          String osName = System.getProperty("os.name" );

*************** public class WindowsViewer extends Defau
*** 73,83 ****
  		}
  		else openDocument( url );

      }

!     public void openWithUrl( String url )
      {
          openDocument( url );
      }


--- 75,87 ----
  		}
  		else openDocument( url );

+         return Child;
      }

!     public Process openWithUrl( String url )
      {
          openDocument( url );
+         return Child;
      }


Index: org/columba/modules/mail/i18n/menu/mainframe.properties
===================================================================
RCS file: /cvsroot/columba/columba/org/columba/modules/mail/i18n/menu/mainframe.properties,v
retrieving revision 1.1
diff -u -3 -p -c -w -r1.1 mainframe.properties
*** org/columba/modules/mail/i18n/menu/mainframe.properties	3 Sep 2002 10:45:33 -0000	1.1
--- org/columba/modules/mail/i18n/menu/mainframe.properties	17 Sep 2002 15:47:42 -0000
*************** menu_file_manage = Manage Messages remot
*** 37,48 ****
  menu_file_printsetup = Printer Setup...


-
  menu_edit_accountconfig = Mail Account Settings...
  menu_edit_preferences = Theme and Font Preferences
  menu_edit_generaloptions = General Options...
  menu_aspell_preferences = Configure ASpell

  menu_edit_searchmessages = Search Messages...
  menu_view_sort = Sort Messages by
  menu_view_next = Next
--- 37,51 ----
  menu_file_printsetup = Printer Setup...


  menu_edit_accountconfig = Mail Account Settings...
  menu_edit_preferences = Theme and Font Preferences
  menu_edit_generaloptions = General Options...
  menu_aspell_preferences = Configure ASpell

+ # 09/16/02 ALP
+ # Added as part of external editor support
+ menu_edit_extern_edit = Use External Editor
+
  menu_edit_searchmessages = Search Messages...
  menu_view_sort = Sort Messages by
  menu_view_next = Next
*************** attachmentsaveas=Save As ...
*** 215,221 ****
  attachmentsaveas_tooltip=Save this Attachment to disk
  attachmentview_mimeheader=View Mime-Header
  attachmentview_mimeheader_tooltip=View the Mime-Header of this Attachment
-
-
-
-
--- 218,220 ----
